### ISSUE TITLE
GUI corrupted after docker installation 

### ISSUE THREAD

**Issue by wastl0815** [inventree/InvenTree#6671]
### Deployment Method

- [ ] Installer
- [ ] Docker Development
- [x] Docker Production
- [ ] Bare metal Development
- [ ] Bare metal Production
- [ ] Digital Ocean image
- [ ] Other (please provide a link `Steps to Reproduce`

### Describe the problem*

I used Portainer to install a stable docker version of the inventree production server. All containers are running stable/healthy. I'm able to login the inventree server and logout. The container seems to work fine. I created and declared the external volumes inside the .env. After installation I used invoke update (inventree-worker) and restarted the container (inventree-server). Everything seems nice and easy...but

https://github.com/inventree/InvenTree/discussions/5053 <-- I used invoke update because of this topic 

Issue: The whole GUI/frontend of Inventree seems to be corrupted after installation. The content of the menu seems to be ok, but no pictures or any visual structur. I'm also unable to add items or make changes in the settings. So the whole function inventree seems to be affected.

### Steps to Reproduce

1. creating folder inventree and subfolders db + data + ext
2. add the subfolders data + ext to the env-file ("external volumes")
3. edit env-file (username, password, etc)
4. deloy a new stack using portainer + the compose-file + env-file
5. after deloyment connect to "inventree-worker" container and use "invoke update"
6. after invoke update, restart "inventree-server" container and login
7. login successful but menu corrupted

all containers are running stable
![portainer_inv](https://github.com/inventree/InvenTree/assets/163018392/4ce28a65-09e1-44f1-9ef4-9686e6c0f23d)

![login_inv](https://github.com/inventree/InvenTree/assets/163018392/c8f4ea8a-0806-4a88-b3da-d016b6dbae11)

![inv_settings](https://github.com/inventree/InvenTree/assets/163018392/64b40029-5184-4760-8e8a-e1ccc2d5cd86)
 

### Relevant log output

```bash
>>> INVOKE UPDATE <<<
/home/inventree # invoke update
Installing required python packages from 'requirements.txt'
Requirement already satisfied: pip in /usr/local/lib/python3.11/site-packages (24.0)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Requirement already satisfied: setuptools in /root/.local/lib/python3.11/site-packages (69.1.0)
Collecting setuptools
  Downloading setuptools-69.1.1-py3-none-any.whl.metadata (6.2 kB)
Downloading setuptools-69.1.1-py3-none-any.whl (819 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 819.3/819.3 kB 3.1 MB/s eta 0:00:00
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 69.1.0
    Uninstalling setuptools-69.1.0:
      Successfully uninstalled setuptools-69.1.0
Successfully installed setuptools-65.5.1
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Requirement already satisfied: asgiref==3.7.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 3)) (3.7.2)
Requirement already satisfied: async-timeout==4.0.3 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 7)) (4.0.3)
Requirement already satisfied: attrs==23.2.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 9)) (23.2.0)
Requirement already satisfied: babel==2.14.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 13)) (2.14.0)
Requirement already satisfied: backoff==2.2.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 15)) (2.2.1)
Requirement already satisfied: bleach==6.1.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 20)) (6.1.0)
Requirement already satisfied: brotli==1.1.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 22)) (1.1.0)
Requirement already satisfied: certifi==2024.2.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 24)) (2024.2.2)
Requirement already satisfied: cffi==1.16.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 28)) (1.16.0)
Requirement already satisfied: charset-normalizer==3.3.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 32)) (3.3.2)
Requirement already satisfied: coreapi==2.3.3 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 34)) (2.3.3)
Requirement already satisfied: coreschema==0.0.4 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 35)) (0.0.4)
Requirement already satisfied: cryptography==42.0.4 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 37)) (42.0.4)
Requirement already satisfied: cssselect2==0.7.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 41)) (0.7.0)
Requirement already satisfied: defusedxml==0.7.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 43)) (0.7.1)
Requirement already satisfied: deprecated==1.2.14 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 47)) (1.2.14)
Requirement already satisfied: diff-match-patch==20230430 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 52)) (20230430)
Requirement already satisfied: dj-rest-auth==5.0.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 54)) (5.0.2)
Requirement already satisfied: django==4.2.10 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 55)) (4.2.10)
Requirement already satisfied: django-allauth==0.60.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 87)) (0.60.1)
Requirement already satisfied: django-allauth-2fa==0.11.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 89)) (0.11.1)
Requirement already satisfied: django-cleanup==8.1.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 90)) (8.1.0)
Requirement already satisfied: django-cors-headers==4.3.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 91)) (4.3.1)
Requirement already satisfied: django-crispy-forms==1.14.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 92)) (1.14.0)
Requirement already satisfied: django-dbbackup==4.1.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 93)) (4.1.0)
Requirement already satisfied: django-error-report-2==0.4.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 94)) (0.4.2)
Requirement already satisfied: django-filter==23.5 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 95)) (23.5)
Requirement already satisfied: django-flags==5.0.13 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 96)) (5.0.13)
Requirement already satisfied: django-formtools==2.5.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 97)) (2.5.1)
Requirement already satisfied: django-ical==1.9.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 98)) (1.9.2)
Requirement already satisfied: django-import-export==3.3.7 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 99)) (3.3.7)
Requirement already satisfied: django-js-asset==2.2.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 100)) (2.2.0)
Requirement already satisfied: django-maintenance-mode==0.21.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 102)) (0.21.1)
Requirement already satisfied: django-markdownify==0.9.3 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 103)) (0.9.3)
Requirement already satisfied: django-money==3.2.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 104)) (3.2.0)
Requirement already satisfied: django-mptt==0.16.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 105)) (0.16.0)
Requirement already satisfied: django-otp==1.3.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 106)) (1.3.0)
Requirement already satisfied: django-picklefield==3.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 108)) (3.1)
Requirement already satisfied: django-q-sentry==0.1.6 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 110)) (0.1.6)
Requirement already satisfied: django-q2==1.6.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 111)) (1.6.1)
Requirement already satisfied: django-recurrence==1.11.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 112)) (1.11.1)
Requirement already satisfied: django-redis==5.4.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 114)) (5.4.0)
Requirement already satisfied: django-sesame==3.2.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 115)) (3.2.2)
Requirement already satisfied: django-sql-utils==0.7.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 116)) (0.7.0)
Requirement already satisfied: django-sslserver==0.22 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 117)) (0.22)
Requirement already satisfied: django-stdimage==6.0.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 118)) (6.0.2)
Requirement already satisfied: django-taggit==5.0.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 119)) (5.0.1)
Requirement already satisfied: django-user-sessions==2.0.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 120)) (2.0.0)
Requirement already satisfied: django-weasyprint==2.2.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 121)) (2.2.2)
Requirement already satisfied: django-xforwardedfor-middleware==2.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 122)) (2.0)
Requirement already satisfied: djangorestframework==3.14.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 123)) (3.14.0)
Requirement already satisfied: djangorestframework-simplejwt==5.3.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 128)) (5.3.1)
Requirement already satisfied: drf-spectacular==0.27.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 129)) (0.27.1)
Requirement already satisfied: dulwich==0.21.7 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 130)) (0.21.7)
Requirement already satisfied: et-xmlfile==1.1.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 131)) (1.1.0)
Requirement already satisfied: feedparser==6.0.11 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 133)) (6.0.11)
Requirement already satisfied: fonttools==4.47.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 134)) (4.47.2)
Requirement already satisfied: googleapis-common-protos==1.62.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 136)) (1.62.0)
Requirement already satisfied: grpcio==1.54.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 140)) (1.54.2)
Requirement already satisfied: gunicorn==21.2.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 142)) (21.2.0)
Requirement already satisfied: html5lib==1.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 143)) (1.1)
Requirement already satisfied: icalendar==5.0.11 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 145)) (5.0.11)
Requirement already satisfied: idna==3.6 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 147)) (3.6)
Requirement already satisfied: importlib-metadata==6.11.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 149)) (6.11.0)
Requirement already satisfied: inflection==0.5.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 154)) (0.5.1)
Requirement already satisfied: itypes==1.2.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 156)) (1.2.0)
Requirement already satisfied: jinja2==3.1.3 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 158)) (3.1.3)
Requirement already satisfied: jsonschema==4.21.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 160)) (4.21.1)
Requirement already satisfied: jsonschema-specifications==2023.12.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 162)) (2023.12.1)
Requirement already satisfied: markdown==3.5.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 164)) (3.5.2)
Requirement already satisfied: markuppy==1.14 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 166)) (1.14)
Requirement already satisfied: markupsafe==2.1.5 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 168)) (2.1.5)
Requirement already satisfied: oauthlib==3.2.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 170)) (3.2.2)
Requirement already satisfied: odfpy==1.4.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 172)) (1.4.1)
Requirement already satisfied: openpyxl==3.1.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 174)) (3.1.2)
Requirement already satisfied: opentelemetry-api==1.22.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 176)) (1.22.0)
Requirement already satisfied: opentelemetry-exporter-otlp==1.22.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 186)) (1.22.0)
Requirement already satisfied: opentelemetry-exporter-otlp-proto-common==1.22.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 187)) (1.22.0)
Requirement already satisfied: opentelemetry-exporter-otlp-proto-grpc==1.22.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 191)) (1.22.0)
Requirement already satisfied: opentelemetry-exporter-otlp-proto-http==1.22.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 193)) (1.22.0)
Requirement already satisfied: opentelemetry-instrumentation==0.43b0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 195)) (0.43b0)
Requirement already satisfied: opentelemetry-instrumentation-django==0.43b0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 201)) (0.43b0)
Requirement already satisfied: opentelemetry-instrumentation-redis==0.43b0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 202)) (0.43b0)
Requirement already satisfied: opentelemetry-instrumentation-requests==0.43b0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 203)) (0.43b0)
Requirement already satisfied: opentelemetry-instrumentation-wsgi==0.43b0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 204)) (0.43b0)
Requirement already satisfied: opentelemetry-proto==1.22.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 206)) (1.22.0)
Requirement already satisfied: opentelemetry-sdk==1.22.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 211)) (1.22.0)
Requirement already satisfied: opentelemetry-semantic-conventions==0.43b0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 215)) (0.43b0)
Requirement already satisfied: opentelemetry-util-http==0.43b0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 222)) (0.43b0)
Requirement already satisfied: packaging==23.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 227)) (23.2)
Requirement already satisfied: pdf2image==1.17.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 229)) (1.17.0)
Requirement already satisfied: pillow==10.2.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 230)) (10.2.0)
Requirement already satisfied: pint==0.21 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 237)) (0.21)
Requirement already satisfied: protobuf==4.25.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 238)) (4.25.2)
Requirement already satisfied: py-moneyed==3.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 242)) (3.0)
Requirement already satisfied: pycparser==2.21 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 244)) (2.21)
Requirement already satisfied: pydyf==0.8.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 246)) (0.8.0)
Requirement already satisfied: pyjwt==2.8.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 248)) (2.8.0)
Requirement already satisfied: pyphen==0.14.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 252)) (0.14.0)
Requirement already satisfied: pypng==0.20220715.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 254)) (0.20220715.0)
Requirement already satisfied: python-barcode==0.15.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 256)) (0.15.1)
Requirement already satisfied: python-dateutil==2.8.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 257)) (2.8.2)
Requirement already satisfied: python-dotenv==1.0.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 261)) (1.0.1)
Requirement already satisfied: python-fsutil==0.13.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 262)) (0.13.1)
Requirement already satisfied: python3-openid==3.2.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 264)) (3.2.0)
Requirement already satisfied: pytz==2024.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 266)) (2024.1)
Requirement already satisfied: pyyaml==6.0.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 271)) (6.0.1)
Requirement already satisfied: qrcode==7.4.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 275)) (7.4.2)
Requirement already satisfied: rapidfuzz==3.6.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 277)) (3.6.1)
Requirement already satisfied: redis==5.0.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 278)) (5.0.1)
Requirement already satisfied: referencing==0.33.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 280)) (0.33.0)
Requirement already satisfied: regex==2023.12.25 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 284)) (2023.12.25)
Requirement already satisfied: requests==2.31.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 285)) (2.31.0)
Requirement already satisfied: requests-oauthlib==1.3.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 291)) (1.3.1)
Requirement already satisfied: rpds-py==0.17.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 293)) (0.17.1)
Requirement already satisfied: sentry-sdk==1.40.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 297)) (1.40.0)
Collecting setuptools==69.1.0 (from -r requirements.txt (line 299))
  Downloading setuptools-69.1.0-py3-none-any.whl.metadata (6.1 kB)
Requirement already satisfied: sgmllib3k==1.0.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 303)) (1.0.0)
Requirement already satisfied: six==1.16.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 305)) (1.16.0)
Requirement already satisfied: sqlparse==0.4.4 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 310)) (0.4.4)
Requirement already satisfied: tablib==3.5.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 314)) (3.5.0)
Requirement already satisfied: tinycss2==1.2.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 316)) (1.2.1)
Requirement already satisfied: typing-extensions==4.9.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 321)) (4.9.0)
Requirement already satisfied: uritemplate==4.1.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 328)) (4.1.1)
Requirement already satisfied: urllib3==2.2.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 332)) (2.2.0)
Requirement already satisfied: weasyprint==60.2 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 337)) (60.2)
Requirement already satisfied: webencodings==0.5.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 339)) (0.5.1)
Requirement already satisfied: wrapt==1.16.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 345)) (1.16.0)
Requirement already satisfied: xlrd==2.0.1 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 350)) (2.0.1)
Requirement already satisfied: xlwt==1.3.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 352)) (1.3.0)
Requirement already satisfied: zipp==3.17.0 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 354)) (3.17.0)
Requirement already satisfied: zopfli==0.2.3 in /root/.local/lib/python3.11/site-packages (from -r requirements.txt (line 356)) (0.2.3)
Downloading setuptools-69.1.0-py3-none-any.whl (819 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 819.3/819.3 kB 7.2 MB/s eta 0:00:00
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 65.5.1
    Uninstalling setuptools-65.5.1:
      Successfully uninstalled setuptools-65.5.1
Successfully installed setuptools-69.1.1
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Installing plugin packages from '/home/inventree/data/plugins.txt'
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Backing up InvenTree database...
Python version 3.11.8 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Backing Up Database: inventree
Writing file to default-8af84e03ddda-2024-03-11-162449.psql.bin.gz
Backing up InvenTree media files...
Python version 3.11.8 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Writing file to 8af84e03ddda-2024-03-11-162453.tar.gz
Running InvenTree database migrations...
========================================
Python version 3.11.8 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Failed to set maintenance mode state
No changes detected
Python version 3.11.8 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Failed to set maintenance mode state
Operations to perform:
  Apply all migrations: InvenTree, account, admin, auth, build, common, company, contenttypes, django_q, error_report, exchange, flags, label, machine, order, otp_static, otp_totp, part, plugin, report, socialaccount, stock, taggit, user_sessions, users
Running migrations:
  Applying InvenTree.0001_initial... OK
  .
  . etc ...
  .
  Applying user_sessions.0003_auto_20161205_1516... OK
Maintenance mode was not disabled - forcing it now
Python version 3.11.8 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Operations to perform:
  Synchronize unmigrated apps: allauth, allauth_2fa, corsheaders, crispy_forms, dbbackup, dj_rest_auth, django_cleanup, django_filters, django_ical, django_otp, djmoney, drf_spectacular, formtools, generic, import_export, maintenance_mode, markdownify, messages, mptt, registration, rest_framework, staticfiles, web
  Apply all migrations: InvenTree, account, admin, auth, build, common, company, contenttypes, django_q, error_report, exchange, flags, label, machine, order, otp_static, otp_totp, part, plugin, report, socialaccount, stock, taggit, user_sessions, users
Synchronizing apps without migrations:
  Creating tables...
    Running deferred SQL...
Running migrations:
  No migrations to apply.
========================================
InvenTree database migrations completed!
Skipping frontend update!
Python version 3.11.8 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Rendered all files in /home/inventree/InvenTree/templates/js/translated
Collecting static files...
Python version 3.11.8 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Redefining 'celsius' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
Redefining 'fahrenheit' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
Redefining 'kelvin' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
Python version 3.11.8 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Redefining 'celsius' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
Redefining 'fahrenheit' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
Redefining 'kelvin' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
Python version 3.11.8 - /usr/local/bin/python3
File “/home/inventree/InvenTree/locale/nl/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/he/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/fr/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/id/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/sr/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/it/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/zh_Hans/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/el/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/bg/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/pt/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/pt_br/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/sv/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/en/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/da/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/ko/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/fi/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/hi/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/ja/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/zh/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/th/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/vi/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/es/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/cs/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/zh_hant/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/es_MX/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/fa/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/tr/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/ru/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/hu/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/no/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/de/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/sk/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/pl/LC_MESSAGES/django.po” is already compiled and up to date.
File “/home/inventree/InvenTree/locale/sl/LC_MESSAGES/django.po” is already compiled and up to date.
InvenTree translation coverage: 24%


>>> INVENTREE-SERVER LOG <<<
   raise ex.with_traceback(None)
django.db.utils.ProgrammingError: relation "django_q_schedule" does not exist
LINE 1: ...", "django_q_schedule"."intended_date_kwarg" FROM "django_q_...
                                                             ^
Traceback (most recent call last):
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/psycopg/cursor.py", line 732, in execute
    raise ex.with_traceback(None)
psycopg.errors.UndefinedTable: relation "django_q_schedule" does not exist
LINE 1: ...", "django_q_schedule"."intended_date_kwarg" FROM "django_q_...
                                                             ^
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/root/.local/bin/gunicorn", line 8, in <module>
    sys.exit(run())
             ^^^^^
  File "/root/.local/lib/python3.11/site-packages/gunicorn/app/wsgiapp.py", line 67, in run
    WSGIApplication("%(prog)s [OPTIONS] [APP_MODULE]").run()
  File "/root/.local/lib/python3.11/site-packages/gunicorn/app/base.py", line 236, in run
    super().run()
  File "/root/.local/lib/python3.11/site-packages/gunicorn/app/base.py", line 72, in run
    Arbiter(self).run()
    ^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/gunicorn/arbiter.py", line 58, in __init__
    self.setup(app)
  File "/root/.local/lib/python3.11/site-packages/gunicorn/arbiter.py", line 118, in setup
    self.app.wsgi()
  File "/root/.local/lib/python3.11/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
                    ^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
           ^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/gunicorn/util.py", line 371, in import_app
    mod = importlib.import_module(module)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/inventree/InvenTree/InvenTree/wsgi.py", line 17, in <module>
    application = get_wsgi_application()  # pragma: no cover
                  ^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/core/wsgi.py", line 12, in get_wsgi_application
    django.setup(set_prefix=False)
  File "/root/.local/lib/python3.11/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/root/.local/lib/python3.11/site-packages/django/apps/registry.py", line 124, in populate
    app_config.ready()
  File "/home/inventree/InvenTree/InvenTree/apps.py", line 54, in ready
    self.start_background_tasks()
  File "/home/inventree/InvenTree/InvenTree/apps.py", line 99, in start_background_tasks
    for existing_task in Schedule.objects.all():
  File "/root/.local/lib/python3.11/site-packages/django/db/models/query.py", line 398, in __iter__
    self._fetch_all()
  File "/root/.local/lib/python3.11/site-packages/django/db/models/query.py", line 1881, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/models/query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "/root/.local/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/psycopg/cursor.py", line 732, in execute
    raise ex.with_traceback(None)
django.db.utils.ProgrammingError: relation "django_q_schedule" does not exist
LINE 1: ...", "django_q_schedule"."intended_date_kwarg" FROM "django_q_...
                                                             ^
Loading config file : /home/inventree/data/config.yaml
Python version 3.11.8 - /usr/local/bin/python
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Redefining 'celsius' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
Redefining 'fahrenheit' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
Redefining 'kelvin' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)


>>> INVENTREE-WORKER LOG <<<
   return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/psycopg/cursor.py", line 732, in execute
    raise ex.with_traceback(None)
django.db.utils.ProgrammingError: column django_q_schedule.cron does not exist
LINE 1: ...edule"."repeats", "django_q_schedule"."next_run", "django_q_...
                                                             ^
Could not create task from schedule
Traceback (most recent call last):
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/psycopg/cursor.py", line 732, in execute
    raise ex.with_traceback(None)
psycopg.errors.UndefinedColumn: column django_q_schedule.cron does not exist
LINE 1: ...edule"."repeats", "django_q_schedule"."next_run", "django_q_...
                                                             ^
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/root/.local/lib/python3.11/site-packages/django_q/scheduler.py", line 41, in scheduler
    for s in (
  File "/root/.local/lib/python3.11/site-packages/django/db/models/query.py", line 398, in __iter__
    self._fetch_all()
  File "/root/.local/lib/python3.11/site-packages/django/db/models/query.py", line 1881, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/models/query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "/root/.local/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/psycopg/cursor.py", line 732, in execute
    raise ex.with_traceback(None)
django.db.utils.ProgrammingError: column django_q_schedule.cron does not exist
LINE 1: ...edule"."repeats", "django_q_schedule"."next_run", "django_q_...
                                                             ^
Could not create task from schedule
Traceback (most recent call last):
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/psycopg/cursor.py", line 732, in execute
    raise ex.with_traceback(None)
psycopg.errors.UndefinedColumn: column django_q_schedule.cron does not exist
LINE 1: ...edule"."repeats", "django_q_schedule"."next_run", "django_q_...
                                                             ^
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/root/.local/lib/python3.11/site-packages/django_q/scheduler.py", line 41, in scheduler
    for s in (
  File "/root/.local/lib/python3.11/site-packages/django/db/models/query.py", line 398, in __iter__
    self._fetch_all()
  File "/root/.local/lib/python3.11/site-packages/django/db/models/query.py", line 1881, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/models/query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1562, in execute_sql
    cursor.execute(sql, params)
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    with self.db.wrap_database_errors:
  File "/root/.local/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/root/.local/lib/python3.11/site-packages/django/db/backends/utils.py", line 89, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/.local/lib/python3.11/site-packages/psycopg/cursor.py", line 732, in execute
    raise ex.with_traceback(None)
django.db.utils.ProgrammingError: column django_q_schedule.cron does not exist
LINE 1: ...edule"."repeats", "django_q_schedule"."next_run", "django_q_...
                                                             ^
/root/.local/lib/python3.11/site-packages/django/db/models/fields/__init__.py:1535: RuntimeWarning: DateTimeField PartPricing.updated received a naive datetime (2024-02-10 00:00:00) while time zone support is active.
  warnings.warn(
No active currency exchange plugins found - skipping update
/root/.local/lib/python3.11/site-packages/django/db/models/fields/__init__.py:1595: RuntimeWarning: DateTimeField NotesImage.date received a naive datetime (2023-12-12 16:34:19.905038) while time zone support is active.
```

**Comment by matmair** [comment]
@wastl0815 seems like your instance is missing some migrations. Could you try running the `invoke update` command on the server container and publish the logs here? Seems like the update script is not reaching the point where it copies the static files

**Comment by wastl0815** [comment]
sorry for the huge font....dont know why is is converted that way

**Comment by wastl0815** [comment]
you need the compose and the env files as well?

**Comment by matmair** [comment]
@wastl0815 I do not think we need the other files right now. Is this the full log output? It is missing a section that would be required for static files to work.

please try running this instead `invoke static`

**Comment by wastl0815** [comment]
yes, this the whole log file when using invoke update on the server container

I used invoke static instead: log-file below

/home/inventree # invoke static
Python version 3.11.8 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Rendered all files in /home/inventree/InvenTree/templates/js/translated
Collecting static files...
Python version 3.11.8 - /usr/local/bin/python3
/root/.local/lib/python3.11/site-packages/allauth/exceptions.py:9: UserWarning: allauth.exceptions is deprecated, use allauth.core.exceptions
  warnings.warn("allauth.exceptions is deprecated, use allauth.core.exceptions")
Redefining 'celsius' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
Redefining 'fahrenheit' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
Redefining 'kelvin' (<class 'pint.delegates.txt_defparser.plain.UnitDefinition'>)
/home/inventree #

**Comment by matmair** [comment]
@wastl0815 ist it working now? These logs seem ordinary.  Are you using the default setup for docker or have you changed anything (except the .env file - that is expected)

**Comment by wastl0815** [comment]
yes, i had to change the original compose file from the inventree website because when using the original file error messages occur on portainer/stack. I wasn't able to get it running in its original form.

please see the modified compose file below:

version: "3.9"
services:
  inventree-db:
    image: postgres:13
    container_name: inventree-db
    hostname: inventree-db
    mem_limit: 1g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "pg_isready", "-q", "-d", "inventree", "-U", "inventreeuser"]
      timeout: 45s
      interval: 10s
      retries: 10
    volumes:
      - /volume1/docker/inventree/db:/var/lib/postgresql/data:rw
    environment:
      POSTGRES_DB: inventree
      POSTGRES_USER: inventreeuser
      POSTGRES_PASSWORD: inventreepass
    restart: on-failure:5

  inventree-cache:
    image: redis:7.0
    container_name: inventree-cache
    depends_on:
      inventree-db:
        condition: service_healthy
    env_file:
      - stack.env
    restart: on-failure:5

  inventree-server:
    image: inventree/inventree:stable
    container_name: inventree-server
    ports:
      - 5234:8000
    depends_on:
      inventree-db:
        condition: service_healthy
    env_file:
      - stack.env
    volumes:
      - /volume1/docker/inventree/data:/home/inventree/data:rw
      - /volume1/docker/inventree/ext:/home/inventree/ext:rw
    restart: on-failure:5

  inventree-worker:
    image: inventree/inventree:stable
    container_name: inventree-worker
    command: invoke worker
    depends_on:
      inventree-db:
        condition: service_healthy
    env_file:
      - stack.env
    volumes:
      - /volume1/docker/inventree/data:/home/inventree/data:rw
      - /volume1/docker/inventree/ext:/home/inventree/ext:rw
    restart: on-failure:5

And the stack.env:

INVENTREE_EXT_VOLUME=/home/inventree/data
INVENTREE_EXT_VOLUME=/home/inventree/ext
INVENTREE_DEBUG=False
INVENTREE_LOG_LEVEL=WARNING
INVENTREE_ADMIN_USER=admin
INVENTREE_ADMIN_PASSWORD=adminPW
INVENTREE_ADMIN_EMAIL=test@test.com
INVENTREE_SITE_URL=http://192.168.2.108
INVENTREE_DB_ENGINE=postgresql
INVENTREE_DB_NAME=inventree
INVENTREE_DB_HOST=inventree-db
INVENTREE_DB_PORT=5432
INVENTREE_DB_USER=inventreeuser
INVENTREE_DB_PASSWORD=inventreepass
INVENTREE_CACHE_HOST=inventree-cache
INVENTREE_CACHE_PORT=6379
INVENTREE_GUNICORN_TIMEOUT=90
INVENTREE_PLUGINS_ENABLED=True
INVENTREE_AUTO_UPDATE=False
INVENTREE_TAG=stable
COMPOSE_PROJECT_NAME=inventree

**Comment by matmair** [comment]
@wastl0815 I do not see a proxy in there?

**Comment by matmair** [comment]
I mean this bit of infrastructure specifically: https://docs.inventree.org/en/stable/start/docker/#file-server

**Comment by wastl0815** [comment]
ah you are right....thank you

i added the proxy to the compose file but unfortunately the proxy container crashes with some connectivity errors:

failed to deploy a stack: Network inventree_wuekro_default Creating Network inventree_wuekro_default Created Container inventree-db Creating Container inventree-db Created Container inventree-worker Creating Container inventree-cache Creating Container inventree-server Creating Container inventree-worker Created Container inventree-cache Created Container inventree-server Created Container inventree-proxy Creating Container inventree-proxy Created Container inventree-db Starting Container inventree-db Started Container inventree-db Waiting Container inventree-db Waiting Container inventree-db Waiting Container inventree-db Healthy Container inventree-db Healthy Container inventree-server Starting Container inventree-cache Starting Container inventree-db Healthy Container inventree-worker Starting Container inventree-cache Started Container inventree-worker Started Container inventree-server Started Container inventree-proxy Starting Error response from daemon: driver failed programming external connectivity on endpoint inventree-proxy (47f7d7c98e371646c22d8b5b232883975b8c961bed260cc7897725357fda7b62): Error starting userland proxy: listen tcp4 0.0.0.0:443: bind: address already in use

 The Caddyfile.txt is stored in the /volume1/docker/inventree/ folder. 

version: "3.9"
services:
  inventree-db:
    image: postgres:13
    container_name: inventree-db
    hostname: inventree-db
    mem_limit: 1g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "pg_isready", "-q", "-d", "inventree", "-U", "inventreeuser"]
      timeout: 45s
      interval: 10s
      retries: 10
    volumes:
      - /volume1/docker/inventree/db:/var/lib/postgresql/data:rw
    environment:
      POSTGRES_DB: inventree
      POSTGRES_USER: inventreeuser
      POSTGRES_PASSWORD: inventreepass
    restart: on-failure:5

  inventree-cache:
    image: redis:7.0
    container_name: inventree-cache
    depends_on:
      inventree-db:
        condition: service_healthy
    env_file:
      - stack.env
    restart: on-failure:5

  inventree-server:
    image: inventree/inventree:stable
    container_name: inventree-server
    ports:
      - 5234:8000
    depends_on:
      inventree-db:
        condition: service_healthy
    env_file:
      - stack.env
    volumes:
      - /volume1/docker/inventree/data:/home/inventree/data:rw
     # - /volume1/docker/inventree/ext:/home/inventree/ext:rw
    restart: on-failure:5

  inventree-worker:
    image: inventree/inventree:stable
    container_name: inventree-worker
    command: invoke worker
    depends_on:
      inventree-db:
        condition: service_healthy
    env_file:
      - stack.env
    volumes:
      - /volume1/docker/inventree/data:/home/inventree/data:rw
      # - /volume1/docker/inventree/ext:/home/inventree/ext:rw
    restart: on-failure:5

  inventree-proxy:
    container_name: inventree-proxy
    image: caddy:alpine
    restart: always
    depends_on:
      - inventree-server
    ports:
      - 80:80
      - 443:443
    env_file:
      - stack.env
    volumes:
      - ./Caddyfile:/etc/caddy/Caddyfile:ro
      - /volume1/docker/inventree/data/static:/var/www/static:z
      - /volume1/docker/inventree/data/media:/var/www/media:z
      - /volume1/docker/inventree/data:/var/log:z
      - /volume1/docker/inventree/data:/data:z
      - /volume1/docker/inventree/data:/config:z



caddyfile

(log_common) {
    log {
        output file /var/log/caddy/{args.0}.access.log
    }
}

{$INVENTREE_SITE_URL:http://192.168.2.108} {
    import log_common inventree

    encode gzip

    request_body {
        max_size 100MB
    }

    handle_path /static/* {
        root * /var/www/static
        file_server
    }

    handle_path /media/* {
        forward_auth {$INVENTREE_SERVER:"http://192.168.2.108:8000"} {
            uri /auth/
        }
        root * /var/www/media
        file_server
    }

    reverse_proxy {$INVENTREE_SERVER:"http://192.168.2.108:8000"}
}

**Comment by matmair** [comment]
It looks like the port 443 is already in use - probably by another web server. If you plan to only run via HTTP, not https you can remove the 443:443 line from your proxy.

**Comment by SchrodingersGat** [comment]
In your Caddyfile:

these lines:

```
 reverse_proxy {$INVENTREE_SERVER:"http://192.168.2.108:8000"}
```

should be 
```
    reverse_proxy {$INVENTREE_SERVER:"http://inventree-server:8000"}
```

Within the docker container, `192.168.2.108` is not an accessible host

**Comment by wastl0815** [comment]
Thank you. You are right Port 443 was blocked by a hidden service. I had to change to port 81 because port 80 was als blocked by another application @matmair Thank you

I also change the caddy file @SchrodingersGat Thank you

It seems that a still have a bit of a mounting/bind problem with the Caddyfile

version: "3.9"
services:
  inventree-db:
    image: postgres:13
    container_name: inventree-db
    hostname: inventree-db
    mem_limit: 1g
    cpu_shares: 1024
    security_opt:
      - no-new-privileges:true
    healthcheck:
      test: ["CMD", "pg_isready", "-q", "-d", "inventree", "-U", "inventreeuser"]
      timeout: 45s
      interval: 10s
      retries: 10
    volumes:
      - /volume1/docker/inventree/db:/var/lib/postgresql/data:rw
    environment:
      POSTGRES_DB: inventree
      POSTGRES_USER: inventreeuser
      POSTGRES_PASSWORD: inventreepass
    restart: on-failure:5

  inventree-cache:
    image: redis:7.0
    container_name: inventree-cache
    depends_on:
      inventree-db:
        condition: service_healthy
    env_file:
      - stack.env
    restart: on-failure:5

  inventree-server:
    image: inventree/inventree:stable
    container_name: inventree-server
    ports:
      - 5234:8000
    depends_on:
      inventree-db:
        condition: service_healthy
    env_file:
      - stack.env
    volumes:
      - /volume1/docker/inventree/data:/home/inventree/data:rw

    restart: on-failure:5

  inventree-worker:
    image: inventree/inventree:stable
    container_name: inventree-worker
    command: invoke worker
    depends_on:
      inventree-db:
        condition: service_healthy
    env_file:
      - stack.env
    volumes:
      - /volume1/docker/inventree/data:/home/inventree/data:rw
    restart: on-failure:5

  inventree-proxy:
    container_name: inventree-proxy
    image: caddy:alpine
    restart: always
    depends_on:
      - inventree-server
    ports:
      - 81:81
    env_file:
      - stack.env
    volumes:
      - ./Caddyfile:/etc/caddy/Caddyfile:ro
      #- /volume1/docker/inventree/etc/caddy/Caddyfile:/etc/caddy/Caddyfile/Caddyfile.txt:ro
      - /volume1/docker/inventree/data/static:/var/www/static:z
      - /volume1/docker/inventree/data/media:/var/www/media:z
      - /volume1/docker/inventree/data:/var/log:z
      - /volume1/docker/inventree/data:/data:z
      - /volume1/docker/inventree/data:/config:z



errormessage:

Failed to deploy a stack:
Network inventree_default Creating
Network inventree_default Created
Container inventree-db Creating
Container inventree-db Created
Container inventree-server Creating
Container inventree-cache Creating
Container inventree-worker Creating
Container inventree-worker Created
Container inventree-server Created
Container inventree-proxy Creating
Container inventree-cache Created
Container inventree-proxy Created
Container inventree-db Starting
Container inventree-db Started
Container inventree-db Waiting
Container inventree-db Waiting
Container inventree-db Waiting
Container inventree-db Healthy
Container inventree-db Healthy
Container inventree-server Starting
Container inventree-cache Starting
Container inventree-db Healthy
Container inventree-worker Starting
Container inventree-cache Started
Container inventree-worker Started
Container inventree-server Started
Container inventree-proxy Starting
Error response from daemon: Bind mount failed: '/data/compose/11/Caddyfile' does not exists

How to mount the caddyfile and where to but it?

**Comment by wastl0815** [comment]
If he Caddyfile is stored in the "Caddyfile" folder and is in the same folder as the compose file and the env file and i use this command

volumes:
  - ./Caddyfile:/etc/caddy/Caddyfile:ro

I get this error message:
Error response from daemon: Bind mount failed: '/data/compose/11/Caddyfile' does not exists

Else the Caddyfile is stored in a different folder as the compose file and i use this command

volumes:
  - /volume1/docker/inventree/Caddyfile:/etc/caddy/Caddyfile:ro
 
i get the error message below

"/volume1/docker/inventree/Caddyfile" to rootfs at "/etc/caddy/Caddyfile": mount /volume1/docker/inventree/Caddyfile:/etc/caddy/Caddyfile (via /proc/self/fd/6), flags: 0x5001: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

**Comment by SchrodingersGat** [comment]
You probably need to flush your mounted docker volumes - after changing things around a few times they may have been corrupted?

**Comment by matmair** [comment]
This sounds like it could be an issue with the SITE_URL setting - are you trying to access your instance under `http://192.168.2.108`?
